back-end/api/rental_listing_api.py

```diff
@@ -16,22 +19,78 @@ def get(self, listing_id=None):
         rentals = RentalListing.query.all()
         return jsonify([rental.serialize for rental in rentals])
     
+    @jwt_required()
+    @swag_from('../static/swagger/rental_post.yml')
     def post(self):
-
+        '''create  a new listing'''
         try:
+            user_id = get_jwt_identity()
             parser = reqparse.RequestParser()
             parser.add_argument('name', type=str, required=True, help='Rental name required')
             parser.add_argument('location', type=str, required=True, help='Location required')
             parser.add_argument('min-rent', type=float, required=True, help='Minimum rent required')
             parser.add_argument('max-rent', type=float, required=True, help='Maximum rent name required')
             parser.add_argument('rented', type=bool, required=True, help='Rent status required')
-            parser.add_argument('owner_id', type=int, required=True, help='Owner ID required')
+            parser.add_argument('owner_id', type=int, default=user_id, required=True, help='Owner ID required')
+            parser.add_argument('images', type=db.JSON, required=False, help='No image added')
+            parser.add_argument('latitude', type=float, required=False, help='Latitude not provided')
+            parser.add_argument('longitude', type=float, required=False, help='Longitude not provided')
 
             args = parser.parse_args()
-            new_rental = RentalListing()
+            new_rental = RentalListing(**args)
+            db.session.add(new_rental)
+            db.session.commit()
```

Comment:
We should think about a db class to abstract the session.add and session.commit into one save() method.
But from next week maybe.